models/classes/Platform/Service/KeyChainGenerator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoLti\models\classes\Platform\Service;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\model\security\Business\Domain\Key\Key;
+use oat\tao\model\security\Business\Domain\Key\KeyChain;
+use oat\taoLti\models\classes\Security\DataAccess\Repository\PlatformKeyChainRepository;
+
+class KeyChainGenerator extends ConfigurableService
+{
+    private const SSL_CONFIG = [
```

Comment:
should we have this config data inside a default config for this service?

Reply:
Do we need to do that this way?

Reply:
it might be good to do this at the platform level , because it will allow you the possibility to change the values from the setup file if in the future we will use different customers with different needs of encryption